logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/MessageContent.kt

```diff
@@ -51,4 +51,6 @@ sealed class MessageContent {
 
     // we can add other types to be processed, but signaling ones shouldn't be persisted
     object Ignored : Signaling() // messages that aren't processed in any way
+
+    data class FailedDecryption(val encodedData: ByteArray? = null) : Regular()
```

Comment:
why did you make it a child of `Regular` and not `Signal` ?

Reply:
Signal messages are not persisted in the DB. In this case, we want to persist these errors may we implement in the future a different way to trigger again the decryption process on them.

Reply:
Maybe because this is a visual message not a signalling message but why is not a System message?